packages/destination-actions/src/destinations/singlestore/send/__tests__/index.test.ts

```diff
@@ -1,90 +1,291 @@
-        }
-      ]
-    }
+  it('send signle payload', async () => {
+    const event = createTestEvent(payload1)
+    nock('https://testhost:445').post('/api/v2/exec', sendSingleJSON).reply(200, {
+      lastInsertId: 1,
+      rowsAffected: 1
+    })
+    const responses = await testDestination.testAction('send', {
+      event,
+      settings,
+      useDefaultMappings: true
+    })
+    expect(responses.length).toBe(1)
+    expect(responses[0].status).toBe(200)
+  })
+
+  it('send multiple payloads', async () => {
+    const events = [createTestEvent(payload1), createTestEvent(payload2), createTestEvent(payload3)]
+    nock('https://testhost:445').post('/api/v2/exec', sendMultipleJSON).reply(200, {
+      lastInsertId: 3,
+      rowsAffected: 3
+    })
+    const responses = await testDestination.executeBatch('send', {
+      events,
+      settings,
+      mapping
+    })
+    expect(responses.length).toBe(3) // ?? Changed this to 3 from 1
+    expect(responses[0].status).toBe(200)
+  })
+
+  it('send failure', async () => {
+    const events = [createTestEvent(payload1), createTestEvent(payload2), createTestEvent(payload3)]
+    nock('https://testhost:445').post('/api/v2/exec', sendMultipleJSON).reply(400, '')
+    await expect(
+      testDestination.executeBatch('send', {
+        events,
+        settings,
+        mapping
+      })
+    ).rejects.toThrow('Failed to insert data: Unknown error')
+  })
+
+  it('send database failure', async () => {
+    const events = [createTestEvent(payload1), createTestEvent(payload2), createTestEvent(payload3)]
+    nock('https://testhost:445').post('/api/v2/exec', sendMultipleJSON).reply(400, 'Database error')
+    await expect(
+      testDestination.executeBatch('send', {
+        events,
+        settings,
+        mapping
+      })
+    ).rejects.toThrow('Failed to insert data: Database error')
+  })
+
+  it('test Authentication successfull', async () => {
+    nock('https://testhost:445').post('/api/v2/exec', createTableJSON).reply(200, {})
+    await testDestination.testAuthentication(settings)
+  })
 
-    const { kafkaConfig, kafkaTopic } = getKafkaConfiguration(settings)
-    const producerRecord = getProducerRecord(kafkaTopic, validPayload)
-    expect(expectedKafkaTopic).toEqual(kafkaTopic)
-    expect(expectedKafkaConfig).toEqual(kafkaConfig)
-    expect(expectedProducerRecord).toEqual(producerRecord)
+  it('test Authentication failure', async () => {
+    nock('https://testhost:445')
+      .post('/api/v2/exec', (body) => {
+        // Parse if string
+        const parsed = typeof body === 'string' ? JSON.parse(body) : body
+        return parsed.database === 'testdb' && parsed.sql.includes('CREATE TABLE')
+      })
```

Comment:
Why can't it be the same as above?
nock('https://testhost:445').post('/api/v2/exec', createTableJSON)